int64 currNode
int64 prevNode
bool currType
